include/slate/BaseMatrix.hh

```diff
@@ -4058,18 +4055,20 @@ std::tuple<int64_t, int64_t, int>
 {
     assert(0 <= i && i < mt());
     assert(0 <= j && j < nt());
-    assert( device == HostNum || (0 <= device && device < num_devices_) );
+    // Given AnyDevice = -3, AllDevices = -2, HostNum = -1,
+    // GPU devices 0, 1, ..., num_devices-1.
+    assert( AnyDevice <= device && device < num_devices_ );
```

Comment:
Does it make sense to get a tuple where the device is not a device_id?

Reply:
I think in the future MatrixStorage will just take `ij_pair`, instead of `ijdev_tuple`, and pass device separately, so we wouldn't need this `global_index` function.